### DISCUSSION TITLE
how to update dropdown Selection options dynamically ?

### DISCUSSION THREAD

**Discussion by morningstarsabrina** [zauberzeug/nicegui#324] [Q&A] [answered]
Hello,

I want to update a dropdown Selection options after the user click on a button with a new list of options but I'm struggling to do this, any ideas ?

Thanks

**Comment by falkoschindler** [comment, marked as answer]
As mentioned [here](https://github.com/zauberzeug/nicegui/discussions/32#discussioncomment-3973570), you can simply manipulate the options of a `ui.select` and call update afterwards:

```py
select = ui.select(['A', 'B', 'C'])

def add_option():
    select.options.append('D')
    select.update()

ui.button('Add option', on_click=add_option)
```

**Reply by morningstarsabrina** [reply]
thanks it work.
I only think its confusing, since there are set_visibility() and set_value() methods so I expected a method to change options.

**Reply by falkoschindler** [reply]
Good point. We should add such method.

**Reply by falkoschindler** [reply]
On second thought, a `set_options` method would only be useful for replacing the whole `options` list or dictionary. For manipulating the existing object this does not help.
But we should add a remark about updating the options in the documentation.

**Reply by morningstarsabrina** [reply]
perfect thanks
